commit 54330a3ff378b915260793b18584bc05cb4a1ee4
Author: Ximin Luo <infinity0@pwned.gg>
Date:   Wed Mar 26 22:54:20 2014 +0000

    portforward.py: make the ProxyServer client connection behaviour overrideable in subclasses

diff --git twisted/protocols/portforward.py twisted/protocols/portforward.py
index 626d5aa..3ecab30 100644
--- twisted/protocols/portforward.py
+++ twisted/protocols/portforward.py
@@ -74,6 +74,10 @@ class ProxyServer(Proxy):
         if self.reactor is None:
             from twisted.internet import reactor
             self.reactor = reactor
+
+        self.connectProxyClient(client)
+
+    def connectProxyClient(self, client)
         self.reactor.connectTCP(self.factory.host, self.factory.port, client)
 
 
